Make automatic gh-pages deploy work

`mkdocs gh-deploy` is not working from Github Action workflow. 
```
ERROR   -  Config value 'markdown_extensions': Failed to load extension 'markdown_include.include'.
```

Another issue could be read and write permission error.

Adding markdown_include to requirements.txt solves this error. But, there is a new error.
```
DEBUG   -  Copying media file: 'reference/DFD_Shrestha.png'
FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/clm-workflow/clm-workflow/docs/reference/DFD_Shrestha.png'
Error: Process completed with exit code 1.
```

--- a/.github/workflows/deploy_doc.yml
+++ b/.github/workflows/deploy_doc.yml
@@ -1,7 +1,10 @@
-name: build
+name: Build and deploy docs
 on:
   push:
     branches:
+      - main
+      - deploy-docs
+    pull_request:
       - main
 jobs:
   deploy:

--- a/docs/reference/DFD_Shrestha.png
+++ b/docs/reference/DFD_Shrestha.png
@@ -1,1 +1,1 @@
-/glade/u/home/amans/tools/clm-workflow/DFD/DFD_Shrestha.png
+../../DFD/DFD_Shrestha.png

--- a/docs/reference/pseudocode.txt
+++ b/docs/reference/pseudocode.txt
@@ -1,1 +1,1 @@
-/glade/u/home/amans/tools/clm-workflow/pseudocode.txt
+../../pseudocode.txt

--- a/docs/tutorial/clm-sim-tutorial.md
+++ b/docs/tutorial/clm-sim-tutorial.md
@@ -1,3 +1,58 @@
 # CLM simulation workflow tutorial
 
+This tutorial walks through the workflow to automate CLM simulations. Please ensure that CTSM is installed. The path where we installed CTSM is called `CTSMDIR`, throughout the tutorial and by `clm-workflow`.
 
+1. CLM simulations are called cases. The manual way to create and run a CLM case would be to:
+    - Create new case.
+    - Run case setup.
+    - Configure the settings that the CLM case will run on.
+    - Build the case.
+    - Submit the case run to a batch job queue.
+2. `clm-workflow` automates all of the above steps. We instruct the workflow how we want the CLM case to be created, configured, and ran. These settings are specified in two files: `caseconfig.cfg` and `xmlconfig.cfg` inside the directory clm_scripts.
+3. First, we will look inside `caseconfig.cfg`. It contains the variables `CASENAME, CTSMDIR, CASEDIR, RES, COMPSET, XMLCONFIG, and ARCHIVE`.
+    - `CASENAME`:
+    - `CTSMDIR`:
+    - `CASEDIR`:
+    - `RES`:
+    - `COMPSET`:
+    - `XMLCONFIG`:
+    - `ARCHIVE`:
+
+```
+CASENAME="name_to_identify_case",
+CTSMDIR="path_to_ctsm",
+CASEDIR="path_to_store_CASENAME",
+RES="CLM_grid_resolution",
+COMPSET="CLM_compset",
+XMLCONFIG="path_to_xmlconfig.cfg",
+ARCHIVE="path_to_CTSM_archive"
+```
+
+4. Derecho requires a mandatory project account to bill compute hour uses. Other machines might also require a project account. So, create an environment variable `PRJ` and store your project account in it. We will create `PRJ` in a `.profile` file in your home directory.
+```
+$ cd ~
+$ echo 'PRJ=YOUR_PROJECT_ACCOUNT;export PRJ'>>.profile
+```
+This will add `PRJ` to an environment variable after a new login. To make it available in the current login, run:
+```
+$ source ~/.profile
+```
+
+5. `xmlconfig.cfg` contains xml configurations which define a CLM run. All possible xmls can be viewed by running `./xmlquery --listall`. It is best to always have xmlconfigs for `RUN_STARTDATE, STOP_N, and STOP_OPTION`. For this tutorial, we will run CLM for 5 days starting 2000-01-01 and let the maximume job wallclock of 30 minutes.
+```
+RUN_STARTDATE=2000-01-01
+STOP_N=5
+STOP_OPTION=ndays
+JOB_WALLCLOCK_TIME=00:30:00
+```
+
+6. Activate our snakemake environment. First, we will load conda module. Then activate our snakemake environment `snakemake_env`.
+```
+$ ml conda
+$ conda activate snakemake_env
+```
+
+7. We are all set up to run the workflow! Just type `snakemake` to run the workflow.
+```
+$ snakemake
+```

--- a/docs/tutorial/index.md
+++ b/docs/tutorial/index.md
@@ -36,6 +36,22 @@ $ ./bin/git-fleximod update
 
 Please go through NCAR's [CTSM tutorial](https://github.com/NCAR/CTSM-Tutorial) for more information on runnning CTSM.
 
+### Installing snakemake 
+
+Follow the steps to install snakemake [here](https://snakemake.readthedocs.io/en/stable/getting_started/installation.html.) 
+
+```
+$ conda create -c conda-forge -c bioconda -n snakemake snakemake
+```
+
+## clm-workflow Installation
+
+Clone the clm-workflow repo [https://github.com/amansnama/clm-workflow](https://github.com/amansnama/clm-workflow) to your machine.
+
+```
+$ git clone https://github.com/amansnama/clm-workflow
+```
+
 ## Workflow Tutorials
 
 - #### [CLM simulation workflow tutorial](clm-sim-tutorial.md)

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 mkdocs
 mkdocstrings[python]
 mkdocs-material
-mkdocs-material-extensions
+markdown-include

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,3 +1,4 @@
 mkdocs
 mkdocstrings[python]
 mkdocs-material
+mkdocs-material-extensions

